fvm/Grid/AnalyticBRadialGridGenerator.cpp

```diff
@@ -24,30 +24,75 @@ using namespace DREAM::FVM;
  */
 
 AnalyticBRadialGridGenerator::AnalyticBRadialGridGenerator(
-     const len_t nr,  real_t r0,  real_t ra, real_t R0, len_t ntheta_ref, len_t ntheta_interp,
-    real_t *rProfiles, len_t nrProfiles, real_t *Gs, real_t *psi_p0s,
-             real_t *kappas, real_t *deltas, real_t *Deltas
-) : RadialGridGenerator(nr), rMin(r0), rMax(ra) {
+     const len_t nr,  real_t r0,  real_t ra, real_t R0, len_t ntheta_interp,
+     struct shape_profiles *profiles
+) : RadialGridGenerator(nr), rMin(r0), rMax(ra), providedProfiles(profiles) {
     this->R0             = R0;
-    this->ntheta_ref     = ntheta_ref;
     this->ntheta_interp  = ntheta_interp;
-    this->GsProvided     = Gs;
-    this->psisProvided   = psi_p0s;
-    this->kappasProvided = kappas;
-    // note: deltas and Deltas should vanish at rProfiles = 0 (physical constraint, fine numerically regardless)
-    this->deltasProvided = deltas;
-    this->DeltasProvided = Deltas;
-    this->nrProfiles     = nrProfiles;
-    this->rProfilesProvided = rProfiles;
+
+    // Find longest vector in 'profiles'
+    struct shape_profiles *pp = profiles;
+
+    auto construct_spline = [](const len_t n, const real_t *r, const real_t *x) {
+        const gsl_interp_type *tp;
+        if (n == 2)
+            tp = gsl_interp_linear;
+        else
+            tp = gsl_interp_steffen;
+
+        gsl_spline *s = gsl_spline_alloc(tp, n);
+        gsl_spline_init(s, r, x, n);
+
+        return s;
+    };
+
+    // Allocate splines for shape parameters (if necessary)
+    if (pp->nG > 1) {
+        this->spline_G = construct_spline(pp->nG, pp->G_r, pp->G);
+        this->gsl_acc_G = gsl_interp_accel_alloc();
```

Comment:
No, because we allow different shape parameters to be given on different radial grids. GSL requires all profiles using the same spline to have exactly the same size.

Reply:
but isn't the gsl_acc object independent of the spline?

Reply:
Not really. The ``gsl_acc`` keeps track of where the spline has been evaluated earlier and which grid points that corresponds to. If we switch the grid which the accelerator is working with all the time we will confuse it and lose the advantage of the accelerator.

Reply:
Oh, I had no clue! I may be using it inappropriately elsewhere; will review.